plugins/manipulator.cpp

```diff
@@ -1253,150 +2309,396 @@ void viewscreen_unitlaborsst::refreshNames()
         }
         if (unit->military.squad_id > -1) {
             cur->squad_effective_name = Units::getSquadName(unit);
-            cur->squad_info = stl_sprintf("%i", unit->military.squad_position + 1) + "." + cur->squad_effective_name;
+            string detail_str = cur->squad_effective_name;
+            if(detail_str.substr(0,4)=="The ")
+                detail_str=detail_str.substr(4);
+
+            cur->squad_info = stl_sprintf("%i", unit->military.squad_position + 1) + "." + detail_str;
         } else {
             cur->squad_effective_name = "";
             cur->squad_info = "";
         }
+
+        if(cur->name == ""){             //to list animals with no name
+            cur->name = cur->profession;
+            cur->profession = "";
+        }
     }
-    calcSize();
+    sizeDisplay();
 }
 
-void viewscreen_unitlaborsst::calcSize()
-{
-    auto dim = Screen::getWindowSize();
-
-    num_rows = dim.y - 11;
-    if (num_rows > units.size())
-        num_rows = units.size();
-
-    int num_columns = dim.x - DISP_COLUMN_MAX - 1;
-
-    // min/max width of columns
-    int col_minwidth[DISP_COLUMN_MAX];
-    int col_maxwidth[DISP_COLUMN_MAX];
-    col_minwidth[DISP_COLUMN_STRESS] = 6;
-    col_maxwidth[DISP_COLUMN_STRESS] = 6;
-    col_minwidth[DISP_COLUMN_SELECTED] = 1;
-    col_maxwidth[DISP_COLUMN_SELECTED] = 1;
-    col_minwidth[DISP_COLUMN_NAME] = 16;
-    col_maxwidth[DISP_COLUMN_NAME] = 16;        // adjusted in the loop below
-    col_minwidth[DISP_COLUMN_DETAIL] = 10;
-    col_maxwidth[DISP_COLUMN_DETAIL] = 10;  // adjusted in the loop below
-    col_minwidth[DISP_COLUMN_LABORS] = 1;
-    col_maxwidth[DISP_COLUMN_LABORS] = NUM_COLUMNS;
-
-    // get max_name/max_prof from strings length
+void viewscreen_unitlaborsst::calcUnitinfoDemands(){
     for (size_t i = 0; i < units.size(); i++)
     {
-        if (col_maxwidth[DISP_COLUMN_NAME] < units[i]->name.size())
-            col_maxwidth[DISP_COLUMN_NAME] = units[i]->name.size();
+        UnitInfo *cur = units[i];
+        cur->demand=0;
 
-        size_t detail_cmp;
-        if (detail_mode == DETAIL_MODE_SQUAD) {
-            detail_cmp = units[i]->squad_info.size();
-        } else if (detail_mode == DETAIL_MODE_JOB) {
-            detail_cmp = units[i]->job_desc.size();
-        } else {
-            detail_cmp = units[i]->profession.size();
+        for (size_t j = 0; j < NUM_LABORS; j++)
+        {
+            df::unit_labor cur_labor = columns[j].labor;
+            df::job_skill cur_skill = columns[j].skill;
+
+            int dm=0;
+            if(cur->unit->status.labors[cur_labor]){
+                if(cur_skill==df::job_skill::NONE){
+                    if(cur_labor==unit_labor::HAUL_STONE
+                     ||cur_labor==unit_labor::HAUL_WOOD
+                     ||cur_labor==unit_labor::HAUL_ITEM
+                     ||cur_labor==unit_labor::HAUL_STONE
+                     ||cur_labor==unit_labor::BUILD_CONSTRUCTION){
+                        dm=12;
```

Comment:
?

Reply:
Maybe this is from difficulty reviewing on your phone. an 'in demand' score is being calculated here.